api/src/main/java/io/opencensus/internal/BaseMessageEvent.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.internal;
+
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.MessageEvent.Type;
+
+/**
+ * Superclass for {@link MessageEvent} and {@link io.opencensus.trace.NetworkEvent} to resolve API
+ * backward compatibility issue.
+ *
+ * @since 0.12.0
+ */
+public abstract class BaseMessageEvent {
```

Comment:
BaseMessageEvent should have a package-private constructor to ensure that there are only two subclasses.